lib/src/utils.dart

```diff
@@ -504,11 +504,27 @@ String prefixLines(String text, {String prefix = '| ', String? firstPrefix}) {
   return lines.join('\n');
 }
 
-/// The subset of strings that don't need quoting in YAML.
-///
-/// This pattern does not strictly follow the plain scalar grammar of YAML,
-/// which means some strings may be unnecessarily quoted, but it's much simpler.
-final _unquotableYamlString = RegExp(r'^[a-zA-Z_-][a-zA-Z_0-9-]*$');
+/// Does this string need quoting in yaml?
+///
+/// This mechanism does not strictly follow the plain scalar grammar of YAML,
+/// which means some strings may be unnecessarily quoted.
+bool _needsYamlQuotes(String s) {
+  // These must be quoted
+  if ([
+    'true', 'True', 'TRUE', //
+    'false', 'False', 'FALSE',
+    'null', 'Null', 'NULL',
+  ].contains(s)) return true;
+  // Numbers must be quoted
+  if (RegExp(r'^[-+]?[0-9]*\.?[0-9]*(e[0-9]+)?$').hasMatch(s)) return true;
```

Comment:
This is a potentially dangerous RegExp because of the `\d*\.?\d*` sequence. It's probably not exponential, but it can be quadratic in the length to reject something like `00000000000000000000000....0000000x`

Consider changing to `\d*(\.\d*)?` (And be aware that it still matches `.` alone.)